Pull request for repo init

Parker-Stafford opened the pull request:
Set up dependencies, server, database and algorithm for creating fake data.

Parker-Stafford committed:
Modified .gitignore to only ignore webpack bundle in dist folder.

Parker-Stafford committed:
Modified readme to contain usage instructions and links to related
repos.

Parker-Stafford committed:
Installed web-pack, react, and babel dependencies.

Parker-Stafford committed:
Set up webpack config file for bundling.

Parker-Stafford committed:
Created index.html main page in dist.

Parker-Stafford committed:
Created first react component to test webpack.

Parker-Stafford committed:
Added express and nodemon to dependencies. Added start script to run
server.

Parker-Stafford committed:
Initialized server.

Parker-Stafford committed:
Added faker as a dependency for generating fake data.

Parker-Stafford committed:
Added mongoose to dependencies.

Parker-Stafford committed:
Imported db into server and connected db to server.

Parker-Stafford committed:
Created database and connection to server. Defined schemas for users and
listings tables.

Parker-Stafford committed:
Implemented algorithm for creating fake data.

emk100 commented on db/db.js:
```diff
@@ -0,0 +1,19 @@
+
+const listingSchema = new mongoose.Schema({
+  id: {type: Number, unique: true},
+  relatedListings: Array
```
It would be more understandable to describe objects inside the array ( maybe in comments)

Parker-Stafford committed:
Moved dist folder to public out of client. Added style.css file. Changed
output path for bundle to public/dist folder.

Parker-Stafford committed:
Changed static file path for serving static files to public/dist.

Parker-Stafford committed:
Added commented out descriptions of each object that will be added to
the relatedListings array in listings table and the favorites array in
users table in DB.
